### PR TITLE
USWDS Beta 3.0 - Component migration phase 3

### LINKED ISSUE #4496: Migration - Phase 3: Create stories for components
## Description

Create stories for the following components:
- [ ] Validation
- [ ] Name form
- [ ] Address form
- [ ] Password reset form
- [ ] Side navigation default
  - [ ] Side navigation one level
  - [ ] Side navigation two levels
  - [ ] Side navigation three levels
  - [ ] Side navigation compare
- [ ] Identifier default
  - [ ] Identifier multiple logos
  - [ ] Identifier no logos
  - [ ] Identifier tax disclaimer
  - [ ] Identifier default Spanish
  - [ ] Identifier multiple logos Spanish
  - [ ] Identifier no logos Spanish
  - [ ] Identifier tax disclaimer Spanish
- [ ] Footer default
  - [ ] Footer big
  - [ ] Footer slim
- [ ] Sign in form

## Additional information

Based on components listed in [spreadsheet 🔒](https://docs.google.com/spreadsheets/d/1iYPFZlIwskbYDKuUlFgH3jfwYkXB41BOS-QvDtBmDZ8/edit#gid=0)

Phase 1: #4370
Phase 2: #4440

### PR DESCRIPTION
**Pull request by amyleadem**
This PR creates storybook listings for the following components. Resolves #4496

- [x]  Validation
- [x]  Name form
- [x]  Address form
- [x]  Password reset form
- [x]  Sign in form
- Side navigation default
- [x]  Side navigation one level
- [x]  Side navigation two levels
- [x]  Side navigation three levels
- Side navigation compare
- [x]  Identifier default
- [x]  Identifier multiple logos
- [x]  Identifier no logos
- [x]  Identifier tax disclaimer
- [x]  Identifier default Spanish
- [x]  Identifier multiple logos Spanish
- [x]  Identifier no logos Spanish
- [x]  Identifier tax disclaimer Spanish
- [x]  Footer default
- [x]  Footer big
- [x]  Footer slim

### CODE REVIEW

**Comment by amyleadem** [comment]
Some notes:
-I left out the items for Side nav compare and side nav default - I figured it was covered by the Level 1-3 stories and their docs. If you'd like to see a compare story, let me know.
- I discovered a few style variances from the component presentation on uswds-site:
    - Forms: text-decoration exists under required asterisk by default, is hidden in uswds-site
    - Identifier component: missing the external link icon that is added to uswds-site
    - Footer component: the width of content is different in each build, I believe it is due to theme setting for $theme-grid-container-max-width (uswds has it set to 'desktop', uswds-site to 'widescreen')
        - Is this a style bug? Should all footer elements align to the same width?
        - Width presentation difference:
        image

**Comment by amyleadem** [comment]
I noticed that on the mailing address form, the assistive/accessibility markup from combo-box.js is printed in the storybook html tab, but does not show under component code in the [docs site](https://designsystem.digital.gov/templates/form-templates/address-form/). 
Will this be a problem?

File: src/templates/usa-sign-in/includes/_value-props.twig [hunk 1]
```diff
@@ -4,15 +4,15 @@
...
+      'description': '<strong>Value proposition 1:</strong> Vivamus nec velit sed leo scelerisque laoreet vestibulum.'
     },
     {
       'img': 'circle-gray-20.svg',
-      'description': '<strong>Value proposition 1:</strong> Vivamus nec velit sed leo scelerisque laoreet vestibulum.>'
+      'description': '<strong>Value proposition 1:</strong> Vivamus nec velit sed leo scelerisque laoreet vestibulum.'
     },
     {
       'img': 'circle-gray-20.svg',
-      'description': '<strong>Value proposition 3:</strong> Vivamus nec velit sed leo scelerisque laoreet vestibulum.>'
```
Comment (mejiaj) [inline src/templates/usa-sign-in/includes/_value-props.twig:15]: Good catch, thanks for fixing these!

**Comment by mejiaj** [comment]
> Some notes:
> 
> * I left out the items for Side nav compare and side nav default - I figured it was covered by the Level 1-3 stories and their docs. If you'd like to see a side nav compare story, let me know.

This is okay for now. We use compare in the [documentation](https://designsystem.digital.gov/components/side-navigation/). We can test this or the `library--develop` branch to see if it's needed or not. 

_If you're curious_ you can install this branch on USWDS-Site based on the instructions [here](https://github.com/uswds/uswds-site#updating-the-uswds-version). You could branch off and update this [PR](https://github.com/uswds/uswds-site/pull/1216).

**Review by mejiaj — CHANGES_REQUESTED** [review]
Great work! 

Just 2 minor comments on placement of an `endif` and the Storybook label for the default footer (call it the default instead of medium in the Story).

There's also some additional investigation needed for that normalize issue.

**Comment by amyleadem** [comment]
@mejiaj I have resolved the comments above, including resolving the normalize conflict by moving the global abbr[] style out of placeholders/forms and into usa-form.scss. Please let me know if you'd rather have that rule globally-scoped instead.

**Review by mejiaj — APPROVED** [review]

**Pull request merged**